src/tokenized-line.coffee

```diff
@@ -321,8 +321,11 @@ class TokenizedLine
     return unless maxColumn?
     return unless @text.length > maxColumn
 
-    if /\s/.test(@text[maxColumn])
-       # search forward for the start of a word past the boundary
+    if isCjkCharacter(@text[maxColumn])
+      # always wrap when a CJK character is at the wrap boundary
+      return maxColumn
```

Comment:
Seems like there would still be an issue if you had latin characters, then CJK characters, then more latin at the wrap boundary. Would we want to use this approach if the line contained any CJK characters at all? I guess you get into a weird state where someone might be writing principally in a latin character set but used a single CJK character for some reason, which this more limited approach avoids.